modules/base/src/state.fz

```diff
@@ -72,13 +78,17 @@ is
   # modify the state, leaving the contents unchanged
   #
   public modify (f S -> S) state T S =>
-    state val (f get) mode.next
+    res := state val (f get) mode.next
+    res.replace
+    res
 
 
   # set state to new, leaving the contents unchanged
   #
   public put (new S) state T S =>
-    state val new mode.next
+    res := state val new mode.next
+    res.replace
+    res
```

Comment:
This is correct, but what puzzles me is that `mode.next` should return `repl` such that the constructor of `oneway_monad`, which is called by the constructor of `state` should already have done a call to `replace`. 

So there seems to be some problem with that mechanism, i.e., the constructor of `oneway_monad` either does not call `replace`, or calls it with the wrong effect type or the old state value.